### DISCUSSION TITLE
v0.1.10

### DISCUSSION THREAD

**Discussion by EdoFro** [EdoFro/Freeplane_Menu-o-Matic#12] [Announcements]
## v0.1.10

* Unpacking powerButtons now gets the labels for icon buttons too
* Added popup menu to remove Tabs
  * Right clicking on tab, (not on toolbars)
* Added vertical separators
  * node text =  `|`
* Improved Panel layout
  * implements Scrollable interface
  * custom ToolbarLayout
* TitledBorder font size can now be defined by user
  * scalling factor (integer between 0 and 200)
  * 100 =  same size as UI font size

<hr /><em>This discussion was created from the release <a href='https://github.com/EdoFro/Freeplane_Menu-o-Matic/releases/tag/v0.1.10'>v0.1.10</a>.</em>

**Comment by EdoFro** [comment]
Now vertical "|" and horizontal "--" separators work different in 'icons only' toolbars:

- horizontal "--" 
   - allways creates a new line of icons
- vertical "|":
   - if the toolbar is wide enough -> all icons stay in the same line
   - if not -> the separator creates a 'new line'

**Reply by EdoFro** [reply]
## Vertical Separators example:
this:
![image](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/9a373ec1-5aad-42f2-beea-6bdc01f54f36)

creates this toolbar if there is enough space:
![image](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/8f7e1078-3382-44e4-a27a-fbff3f4d268e)

but if it gets smaller then the Separators define where the 'new lines' start:
![image](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/9b21a116-6b61-493e-bc73-0bb7ffefcfe6)

## Horizontal Separators example:
this:
![image](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/db3262ce-db03-4a6b-8dbd-d6d54a92121e)

creates this toolbar:
![image](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/9da386df-affe-4f0f-a1ee-954166e07fb2)

**Reply by EdoFro** [reply]
Just to be clear:
the user can use both types of Separators in one toolbar.

**Comment by EdoFro** [comment]
the new ToolbarLayout logic improves how the toolbars behave.

An example using the previous two toolbars:


https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/62902ef7-9312-43cc-a319-f274c692e13a

**Comment by EdoFro** [comment]
Another example with a toolbar that has both kinds of separators:


https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/202bf672-d3cb-4028-b61b-acb4c2e6da8b

**Comment by Illegit** [comment]
I'll start to test it tomorrow.

**Comment by Illegit** [comment]
Please try to manage the menu node called "Node manager" with FP 1.11.7_01, operations sequence:
1) select node
2) unpack it
3) reshape the nodes (if you succeed) adding for example the vertical separator "|"
4) pack it
5) move the packed menu in another position dragging & drop with the mouse
at the end most of the times this node disappears
[log - Copia (2).zip](https://github.com/EdoFro/Freeplane_Menu-o-Matic/files/12528698/log.-.Copia.2.zip)
![Screenshot 2023-09-05 233136](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/62674157/e4d8c487-d947-439c-8225-fae727d0422e)
![Screenshot 2023-09-05 233544](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/62674157/d515b2d5-5794-4033-a93b-1d2a3b353237)
and now do it with FP 1.11.6 same sequence is very normal.
I do not know if there is a problem with the addon or with the new release, please check it and eventually feedback dimitry. (ist a memory problem?)

**Reply by EdoFro** [reply]
this is a problem with FP 1.11.7_01

look [here](https://github.com/freeplane/freeplane/issues/1381)

**Comment by Illegit** [comment]
in my previous even if you move some children of the node they disappear.

**Comment by Illegit** [comment]
but with 1.11.6 everything goes well.

**Comment by Illegit** [comment]
Problem 2:
as you see in this image I have added a new menu of icons called "VSM" but you see just a slight blue bar below and the bar already at the bottom. It means you are not able to see the last bar just a slight top areas of blue color: is it possible to tune the panel scroll bars better?

**Comment by Illegit** [comment]
![image](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/62674157/e38c763a-c620-4407-b8ea-5257c48c73c4)
here the image

**Comment by Illegit** [comment]
just a note: the previous situation happens when the panel remaining height is shorter than the bar size, but if I add 2 menus it readjusts correctly because the 2nd one will be totally out of the panel.

**Reply by EdoFro** [reply]
if you resize de panel (just a little) then it recalculates its size and the scrollbar appears and shows all the toolbars

**Reply by EdoFro** [reply]
I tried to solve this and I couldn't.
But in those cases, if you select another tab and return, then it will be ok.
The same if you close the ToolPanel and reopen it.

I tried to force a recalculation of the panel's size, but it did nothing.

**Reply by Illegit** [reply]
Yes, it is like you write even in my case.

Anyway, I think this menu really nice. Compliments

**Comment by Illegit** [comment]
only one topic: Is it possible not to launch every time we run FP the MoM menu? I haven't found out how to stop it.
Thanks

**Reply by EdoFro** [reply]
It is included into this preference:
`Preferences…->Plugins->Menu-o-Matic->Open auto launch menus`

maybe I should split it into:
- "open auto launch menus on start up"
- "open auto launch menus on map load"

**Comment by Illegit** [comment]
No, sorry I was only indicating the fact that every time starting FP the auto launch runs the Menu-o-Matic menu with the other menus, I ask how to stop this single menu. Just that. Becuase that is a menu that I need seldom not always. Fro the other I think it is ok.

**Reply by EdoFro** [reply]
I think this is an issue for you, because you don't have the icons I use for the menu and if you uncheck the preference `Preferences…->Plugins->Menu-o-Matic->Show buttons labels in Menu-o-Matic toolbar` the bar don't become any smaller.

in my case, with labels:
![image](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/11f7bf57-a29d-4160-9364-6ea30f16687a)

without labels:
![image](https://github.com/EdoFro/Freeplane_Menu-o-Matic/assets/35700575/4353bf25-34c5-4b66-bf4d-92bf9332766a)

I will add custom icons for these buttons in the next version.

After that we can evaluate this issue again.

**Comment by Illegit** [comment]
Ok, understood. Thank you.
